src/internal/eval-tidy.c

```diff
@@ -152,9 +152,30 @@ static bool is_data_mask(sexp* env) {
     r_typeof(env) == r_type_environment &&
     r_env_has(env, data_mask_flag_sym);
 }
-
+sexp* rlang_is_data_mask(sexp* env) {
+  return r_lgl(is_data_mask(env));
+}
 static sexp* data_pronoun_sym = NULL;
 
+static bool data_mask_has(sexp* mask, sexp* sym) {
+  sexp* top_env = r_env_get(mask, data_mask_top_env_sym);
+  if (r_typeof(top_env) != r_type_environment) {
+    r_abort("Corrupt data mask, missing the `.top_env` environment");
+  }
+  if (r_typeof(sym) != r_type_symbol) {
+    r_abort("`sym` must be a symbol");
+  }
+  sexp* env = mask;
+  do {
+    env = r_env_parent(env);
+    if (Rf_findVarInFrame3(env, sym, FALSE) != R_UnboundValue) return true;
```

Comment:
Return should be on a newline so control flow is immediately obvious.

Reply:
You can use `r_env_find()` and `r_missing_sym`.

oh in fact `r_env_has()` will do the check for you.